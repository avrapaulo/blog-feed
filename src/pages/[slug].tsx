import { GetStaticPaths, GetStaticProps } from 'next'
import { motion } from 'framer-motion'
import Head from 'next/head'
import { CommentList } from 'components/comment-list'
import { CommentDetails } from 'components/comment-details'
import { Post } from 'models/post'

interface Props {
  post: Post
}

const variants = {
  hidden: { opacity: 0.5, x: -250 },
  visible: {
    opacity: 1,
    x: 0,
    transition: { delay: 0.1, duration: 0.4 }
  },
  exit: {
    x: '-100vw',
    transition: { ease: 'easeInOut', duration: 1 }
  }
}

export const DetailPage = ({ post }: Props) => (
  <motion.div initial="exit" animate="visible" exit="exit" variants={variants}>
    <Head>
      <title>{post.title}</title>
    </Head>
    <CommentDetails {...post} />
    <CommentList postId={post.id} />
  </motion.div>
)

export const getStaticPaths: GetStaticPaths = async () => {
  const response = await fetch(`${process.env.API_URL}/posts`)
  const posts: Post[] = await response.json()

  const paths = posts.map(({ slug }) => ({
    params: { slug }
  }))

  return {
    paths,
    fallback: false
  }
}

export const getStaticProps: GetStaticProps = async ({ params }) => {
  const id = (params.slug as string).substring(params.slug.lastIndexOf('-') + 1)
  const res = await fetch(`${process.env.API_URL}/posts/${id}`)
  const post = await res.json()

  return { props: { post } }
}

export default DetailPage
