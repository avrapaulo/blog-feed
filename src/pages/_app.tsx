import { AppProps } from 'next/app'
import { Container } from 'react-bootstrap'
import { Header } from 'components/header'
import 'styles/index.scss'

const MyApp = ({ Component, pageProps }: AppProps) => (
  <Container className="md-container">
    <Header />
    <Component {...pageProps} />
  </Container>
)

export default MyApp
