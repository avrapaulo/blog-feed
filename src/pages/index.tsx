import Head from 'next/head'
import { Feed } from 'components/feed'
import { GetServerSideProps } from 'next'
import { Blog } from 'models/blog'

interface Props {
  blogs: Blog[]
}

const HomePage = ({ blogs }: Props) => (
  <>
    <Head>
      <title>Blog Feed</title>
    </Head>
    <Feed blogs={blogs} />
  </>
)

export const getServerSideProps: GetServerSideProps = async () => {
  const response = await fetch(`${process.env.API_URL}/posts`)
  const blogs: Blog[] = await response.json().then(e =>
    e.sort((a: Blog, b: Blog) => {
      if (a.publish_date === b.publish_date) return 0

      return a.publish_date < b.publish_date ? 1 : -1
    })
  )

  return {
    props: {
      blogs
    }
  }
}

export default HomePage
