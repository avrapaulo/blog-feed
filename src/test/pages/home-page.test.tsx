import { render } from 'test/test-utils'
import { HomePage } from 'pages'
import { posts } from 'test/__mocks__/'

describe('Home page', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<HomePage posts={posts} />, {})
    expect(asFragment()).toMatchSnapshot()
  })
})
