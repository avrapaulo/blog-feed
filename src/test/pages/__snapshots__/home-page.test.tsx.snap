// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div>
      <a
        class="link-unstyled"
        href="/slug"
      >
        <div
          class="border-left-0 border-right-0 border-top-0 rounded-0 card"
        >
          <div
            class="card-body"
          >
            <div
              class="card-title h5"
            >
              title
            </div>
            <div
              class="mb-2 text-muted card-subtitle h6"
            >
              February 23, 2020 (10 months ago)
            </div>
            <p
              class="mb-1 card-text"
            >
              description
            </p>
            <div
              class="text-right card-subtitle h6"
            >
              <p>
                <small>
                  author
                </small>
              </p>
            </div>
          </div>
        </div>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
